views/calendar/calendar.js

```diff
@@ -0,0 +1,416 @@
+                            ${AllMonths.Oct[i].year} <a href="https://www.wikitree.com/wiki/${AllMonths.Oct[i].id}">
+                            ${AllMonths.Oct[i].name}</a> ${AllMonths.Oct[i].text} ${
+                            AllMonths.Oct[i].marriage
+                        } [<span class="small"><a href="https://www.wikitree.com/treewidget/${
+                            AllMonths.Oct[i].id
+                        }/6">share tree image</a>]
+                        </span><br>
+                            `);
+                        $(`#Nov`).append(`
+                        <span>${AllMonths.Nov[i].day} ${getMonth(AllMonths.Nov[i].date)}
+                        ${AllMonths.Nov[i].year} <a href="https://www.wikitree.com/wiki/${AllMonths.Nov[i].id}">
+                        ${AllMonths.Nov[i].name}</a> ${AllMonths.Nov[i].text} ${
+                            AllMonths.Nov[i].marriage
+                        } [<span class="small"><a href="https://www.wikitree.com/treewidget/${
+                            AllMonths.Nov[i].id
+                        }/6">share tree image</a>]
+                        </span><br>
+                    `);
+                        $(`#Dec`).append(`
+                        <span>${AllMonths.Dec[i].day} ${getMonth(AllMonths.Dec[i].date)}
+                        ${AllMonths.Dec[i].year} <a href="https://www.wikitree.com/wiki/${AllMonths.Dec[i].id}">
+                        ${AllMonths.Dec[i].name}</a> ${AllMonths.Dec[i].text} ${
+                            AllMonths.Dec[i].marriage
+                        } [<span class="small"><a href="https://www.wikitree.com/treewidget/${
+                            AllMonths.Dec[i].id
+                        }/6">share tree image</a>]
+                        </span><br>
+                    `);
+                    } catch {}
+                }
+            }
+            // And here is just a bunch of random helpers.
+            // This could use some cleanup and simplification.
+            var months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+            function getMonth(date) {
+                if (date.split("-")[0] != "0000" && date.split("-")[1] != "00" && date.split("-")[2] != "00") {
+                    return months[date.split("-")[1] - 1];
+                } else {
+                    return undefined;
+                }
+            }
+
+            $("span[id^='month-']").click(function (e) {
+                $(".calendar").children().removeClass("active box rounded orange").css("display", "none").hide();
+                $(`#${e.target.id.replace("month-", "")}`)
+                    .addClass("active box rounded orange")
+                    .css("display", "block");
+            });
+
+            function pushEvents(person, date, text, link, spouse) {
+                if (getMonth(date) == "Jan" && getMonth(date) != undefined) {
+                    AllMonths.Jan.push({
```

Comment:
No need to do it separately for every month, object properties can be accessed using brackets and keys, not only using dot method.